src/ddo/interfaces/Credentials.ts

```diff
@@ -0,0 +1,16 @@
+export enum CredentialType {
+  address = 'address',
+  credential3Box = 'credential3Box'
+}
+
+export type CredentialAction = 'allow' | 'deny'
+
+export interface Credential {
+  type: CredentialType
+  value: string[]
+}
+
+export interface Credentials {
+  allow?: Credential[]
```

Comment:
I don't really get how this works. You can have a credential in `allow` that has `CredentialType` `deny`.

Reply:
no.  in allow you can have address | 3box.
same for deny